ticket.go

```diff
@@ -126,91 +129,78 @@ func (lw *LibWallet) getTickets(req *GetTicketsRequest) (ticketInfos []*TicketIn
 		return (targetTicketCount > 0) && (len(ticketInfos) >= targetTicketCount), nil
 	}
 
-	var chainClient *rpcclient.Client
-	if n, err := lw.wallet.NetworkBackend(); err == nil {
-		chainClient, _ = chain.RPCClientFromBackend(n)
+	var rpc *dcrd.RPC
+	if n, err := wallet.internal.NetworkBackend(); err == nil {
+		if client, ok := n.(*dcrd.RPC); ok {
+			rpc = client
+		}
 	}
-	if chainClient != nil {
-		err = lw.wallet.GetTicketsPrecise(rangeFn, chainClient, startBlock, endBlock)
+
+	ctx := wallet.shutdownContext()
+	if rpc != nil {
+		err = wallet.internal.GetTicketsPrecise(ctx, rpc, rangeFn, startBlock, endBlock)
 	} else {
-		err = lw.wallet.GetTickets(rangeFn, startBlock, endBlock)
+		err = wallet.internal.GetTickets(ctx, rangeFn, startBlock, endBlock)
 	}
 
 	return
 }
 
 // TicketPrice returns the price of a ticket for the next block, also known as the stake difficulty.
 // May be incorrect if blockchain sync is ongoing or if blockchain is not up-to-date.
-func (lw *LibWallet) TicketPrice(ctx context.Context) (*TicketPriceResponse, error) {
-	sdiff, err := lw.wallet.NextStakeDifficulty()
+func (wallet *Wallet) TicketPrice(ctx context.Context) (*TicketPriceResponse, error) {
+	sdiff, err := wallet.internal.NextStakeDifficulty(ctx)
 	if err == nil {
-		_, tipHeight := lw.wallet.MainChainTip()
+		_, tipHeight := wallet.internal.MainChainTip(ctx)
 		resp := &TicketPriceResponse{
 			TicketPrice: int64(sdiff),
 			Height:      tipHeight,
 		}
 		return resp, nil
 	}
 
-	n, err := lw.wallet.NetworkBackend()
+	n, err := wallet.internal.NetworkBackend()
 	if err != nil {
 		return nil, err
 	}
-	chainClient, err := chain.RPCClientFromBackend(n)
-	if err != nil {
-		return nil, translateError(err)
-	}
 
 	ticketPrice, err := n.StakeDifficulty(ctx)
 	if err != nil {
 		return nil, translateError(err)
 	}
-	_, blockHeight, err := chainClient.GetBestBlock()
-	if err != nil {
-		return nil, translateError(err)
-	}
 
+	_, tipHeight := wallet.internal.MainChainTip(ctx)
 	return &TicketPriceResponse{
 		TicketPrice: int64(ticketPrice),
-		Height:      int32(blockHeight),
+		Height:      int32(tipHeight),
 	}, nil
 }
 
 // PurchaseTickets purchases tickets from the wallet. Returns a slice of hashes for tickets purchased
-func (lw *LibWallet) PurchaseTickets(ctx context.Context, request *PurchaseTicketsRequest) ([]string, error) {
+func (wallet *Wallet) PurchaseTickets(ctx context.Context, request *PurchaseTicketsRequest, vspHost string) ([]string, error) {
 	var err error
 
-	// fetch redeem script, ticket address, pool address and pool fee if vsp host is configured
-	vspHost := lw.ReadStringConfigValueForKey(VSPHostConfigKey)
```

Comment:
Is there a reason we're taking this out? Raedah mentioned that he wanted this purchase ticket feature implemented such that if the user saved a vsp host info in config, it should be used always. It should not be a parameter to be passed in.

Reply:
Config cannot be accessed from here. It can simply be fetched from config and passed to this function.